apps/ejabberd/src/mod_mam_riak_timed_arch_yz.erl

```diff
@@ -174,8 +174,16 @@ lookup_messages_muc(Result, Host,
                     IsSimple).
 
 
-archive_size(Size, _Host, _ArchiveID, _ArchiveJID) ->
-    Size.
+archive_size(_Size, _Host, _ArchiveID, ArchiveJID) ->
+    OwnerJID = ArchiveJID,
+    RemoteJID = undefined,
+    {MsgIdStartNoRSM, MsgIdEndNoRSM} =
+        calculate_msg_id_borders(undefined, undefined, undefined, undefined),
+    F = fun get_msg_id_key/3,
+    {TotalCount, _} = read_archive(OwnerJID, RemoteJID,
```

Comment:
What you need, here is only the parameter `Count` from [1]. This is returned from single call to Yokozuna. There is no need to further go through all results and fetch messages all we need here is only the count, which is available without reading whole messages.

[1]. https://github.com/esl/MongooseIM/blob/d1ce27d2ac9aade7bec5b079041f6975709cb7c1/apps/ejabberd/src/mod_mam_riak_timed_arch_yz.erl#L397